contrib/flo-bigquery/src/main/java/com/spotify/flo/contrib/bigquery/BigQueryMocking.java

```diff
@@ -125,8 +138,26 @@ public boolean tableExists(TableId tableId) {
       return BigQueryMocking.this.tableExists(tableId);
     }
 
+    @Override
+    public StagingTableId createStagingTableId(BigQueryContext context, TableId tableId) {
+      return Optional.ofNullable(stagingTableIds.get(formatTableIdKey(tableId)))
+          //read from mocked stagingTableIds if exists
+          .map(stagingTableId -> StagingTableId.of(context, stagingTableId))
+          //create a new StagingTableId
+          .orElse(StagingTableId.of(
+              context,
+              TableId.of(
+                  tableId.getProject(),
+                  "_incoming_test",
+                  "_" + tableId.getTable() + "_" + ThreadLocalRandom.current().nextLong(10_000_000))
```

Comment:
This random table name creation should've been in `DefaultBigQueryClient.createStagingTableId()` as well.